src/Statistics.jl

```diff
@@ -0,0 +1,116 @@
+end
+mean(M::Euclidean, x, w) = mean(x,w)
+mean(M::Euclidean, x) = mean(x)
+function mean!(M::Manifold, y, x::AbstractVector, w::AbstractWeights = weights(ones(length(x)) / length(x));
+            stop_iter=100,
+            kwargs...
+        ) where {T}
+    yold = y
+    v = zero_tangent_vector.(Ref(M), fill(y,length(x)))
+    for i=1:stop_iter
+        yold = y
+        log!.(Ref(M), v, Ref(yold), x)
+        y = exp(M, yold, sum( values(w).*v ) / 2 )
+        isapprox(M,y,yold; kwargs...) && break
+    end
+    return y
+end
+
+@doc doc"""
+    median(M,x; weights=1/n*ones(n), stop_tol=10^-10, stop_iter=10000, use_rand = false )
+
+computes the Riemannian median of the vector `x`  of `n` points on the
+[`Manifold`](@ref) `M`. This function is nonsmooth (i.e nondifferentiable) and
+uses a cyclic proximal point scheme. Optionally one can provide
+weights $w_i$ for the weighted Riemannian median. The general formula to compute
+the minimizer reads
+````math
+\argmin_{y\in\mathcal M}\sum_{i=1}^n w_i\mathrm{d}_{\mathcal M}(y,x_i),
+````
+where $\mathrm{d}_{\mathcal M}$ denotes the Riemannian [`distance`](@ref).
+
+The cyclic proximal point can be run with a random cyclic order by setting
+`use_rand` to `true`
+
+Finally `stop_iter` denotes the maximal number of iterations to perform and
+the `kwargs...` are passed to [`isapprox`](@ref) to stop, when the inimal change
+between two iterates is small. For more stopping criteria
+check the [`Manopt.jl`](https://manoptjl.org) package and use a solver therefrom.
+
+The algorithm is further described in Algorithm 4.3 and 4.4 in 
+> Bačák, M: Computing Medians and Means in Hadamard Spaces.
+> SIAM Journal on Optimization (2014), 24(3), pp. 1542–1566,
+> doi: [10.1137/140953393](https://doi.org/10.1137/140953393),
+> arxiv: [1210.2145](https://arxiv.org/abs/1210.2145)
+"""
+function median(M::Manifold, x::AbstractVector, w::AbstractWeights = weights(ones(length(x)) / length(x)); kwargs...)
+    y = x[1]
+    return median!(M, y, x; kwargs...)
+end
+function median!(M::Manifold, y, x::AbstractVector, w::AbstractWeights = weights(ones(length(x)) / length(x));
+            stop_iter=10000,
+            use_random = false,
+            kwargs...
+        ) where {T}
+    n = length(x)
+    yold = y
+    order = collect(1:n)
+    iter = 0
+    for i=1:stop_iter
+        λ = 1/(iter+1)
+        yold = y
+        use_random && shuffle!(order)
+        v = zero_tangent_vector(M,y)
+        for i=1:n
+            t = min( λ * w[order[i]] / distance(M,y,x[order[i]]) , 1 )
+            exp!( M, y, yold, t*log!(M, v, y, x[order[i]]) )
+        end
+        isapprox(M,y,yold; kwargs...) && break
+    end
+    return y
+end
+median(M::Euclidean,x) = median(x)
+median(M::Euclidean,x,w) = median(x,w)
+@doc doc"""
+    var(M,x, corrected::Bool=true, mean=mean(M,x))
+
+compute the variance of the points in `x` on the [`Manifold`](@ref) `M`, i.e.
+````
+    \frac{1}{m}\sum_{i=1}^n d_{\mathcal M}(m,x_i),
```

Comment:
I think that should be the squared distance? And you have two different things called `m`.

Reply:
Thanks for noticing, I switched for the normalise to `n'` in the doc..